packages/frontend/src/hooks/useApi.ts

```diff
@@ -1,20 +1,140 @@
 import useSWR from 'swr';
+import { paths } from '../types/api-schema';
 
-interface UseApiArgs {
+interface UseApiArgs<
+  TUrl extends keyof paths,
+  TMethod extends keyof paths[TUrl],
+  TBodyContentType extends string = 'application/json',
+  TResponseContentType extends string = 'application/json',
+> {
   /** The method, i.e. GET, POST */
-  method?: string;
+  method?: TMethod;
   /** The body of the request */
```

Comment:
should we consider to remove comments to the self-documenting code for a better readability? For instance, the variable name "body" in the context of API request is already self-telling that the variable is for the request body.

Reply:
Possibly, but this is mainly for IDE type suggestions. `fetch` has comments for everything so I was just basing this off that since most of these are passed onto the fetch call.

I can remove them if wanted.

Reply:
Ok, I guess that's fine.